### DISCUSSION TITLE
Maximum possible C++ standard compatible with the library

### DISCUSSION THREAD

**Discussion by fortwoone** [nlohmann/json#4238] [Q&A] [answered]
Hey there,

I see C++11 is mentioned as being only "sparsely" supported by this library. Does it mean I have to use an earlier C++ standard to be able to fully use the library? If not so, what's the most recent C++ standard I can use with the library?

**Comment by gregmarr** [comment, marked as answer]
Where do you see that mentioned?  I can't find it anywhere.  Anyway, this library works in C++11 or later, and even has some support for later features such as `std::string_view` and heterogeneous comparisons and the spaceship operator.

**Reply by fortwoone** [reply]
Thanks for clearing that out! Actually, it’s said right here in the README (section Supported compilers) : https://github.com/nlohmann/json#supported-compilers

**Reply by gregmarr** [reply]
That's referring to *compiler* support, not this library's support.

**Reply by fortwoone** [reply]
Ooooh sorry. Thansk for taking some time to answer, then!
